sound/soc/intel/boards/sof_sdw_rt1308.c

```diff
@@ -130,7 +130,7 @@ int sof_sdw_rt1308_init(const struct snd_soc_acpi_link_adr *link,
 			struct sof_sdw_codec_info *info,
 			bool playback)
 {
-	info->amp_num++;
+	info->amp_num += dai_links->num_codecs;
 	if (info->amp_num == 1)
 		dai_links->init = first_spk_init;
```

Comment:
@bardliao How about the logic below checking "info->amp_num == 1". Is this still valid in case amp_num is incremented possibly by many amps in one call?

Reply:
Agree with @kv2019i . This code assumes amp_num ==1 or ==2, I don't understand this at all. Also num_codecs can include other devices, not just the amps, so the logic is not future proof?